File: tools/test_format.py
#!/usr/bin/env python
# This program fails if ./tools/format.py changes any files.

import sys
import subprocess
import util


def main():
    util.run([sys.executable, "tools/format.py"])
    result = util.run_output(
        ["git", "status", "-uno", "--porcelain", "--ignore-submodules"],
        exit_on_fail=True)
    if result.out:
        print "Run tools/format.py "
        print result.out
        sys.exit(1)


if __name__ == '__main__':
    main()

File: tools/test_format.ts
#!/usr/bin/env deno --allow-run
// This program fails if ./tools/format.ts changes any files.

import { cwd, exit, RunOptions } from "deno";
import {
  findFiles,
  ProcessOptions,
  resolveProcess,
  ProcessResult
} from "./third_party.ts";

async function main() {
  // Format code
  await import("./format.ts");

  // Check for git changes
  const { message, success, stdout } = await resolveProcess([
    "git status",
    {
      args: ["git", "status", "-uno", "--porcelain", "--ignore-submodules"]
    }
  ]);

  if (await stdout) {
    console.log(message);
    console.log("✖ files changed");
    console.log(await stdout);
    exit(1);
  }

  if (message) {
    return Promise.reject(message);
  }

  if (!success) {
    return Promise.reject("git process failed");
  }
}

main();
